meilisearch/task.py

```diff
@@ -11,158 +11,161 @@
 from meilisearch.models.task import TaskInfo
 
 
-def get_tasks(
-    config: Config, parameters: Optional[Dict[str, Any]] = None
-) -> Dict[str, List[Dict[str, Any]]]:
-    """Get all tasks.
-
-    Parameters
-    ----------
-    config:
-        Config object containing permission and location of Meilisearch.
-    parameters (optional):
-        parameters accepted by the get tasks route: https://docs.meilisearch.com/reference/api/tasks.html#get-tasks.
-
-    Returns
-    -------
-    task:
-        Dictionary with limit, from, next and results containing a list of all enqueued, processing, succeeded or failed tasks.
-
-    Raises
-    ------
-    MeiliSearchApiError
-        An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+class TaskHandler:
     """
-    http = HttpRequests(config)
-    if parameters is None:
-        parameters = {}
-    for param in parameters:
-        if isinstance(parameters[param], list):
-            parameters[param] = ",".join(parameters[param])
-    return http.get(f"{config.paths.task}?{parse.urlencode(parameters)}")
-
+    This class handles all tasks related operations for Meilisearch.
```

Comment:
For consistency with the  with the other classes lets split this and move the parameters in the the `__init__`.  I think we can also drop `http` from the doc string since it is really a private variable to the class.

```py
class TaskHandler:
    """This class handles all tasks related operations for Meilisearch."""

    def __init__(self, config: Config):
    """"
    Parameters
    ----------
    config: Config object containing permission and location of Meilisearch.
    """"
    ...
```

Reply:
If we're going with numpy style docstrings shouldn't we stick to the standard.

Reagarding `http` , if we're really keeping it private should I replace it with `__http`?

Reply:
As far as I know the numpy style isn't officially being used. @alallema can tell us if this is the case. I was just matching up to the other docstings here.

Personally I would just go with `_http` for the private, however everywhere else in the package it is set as `http` so just to keep things consistent I would stay with `http`

Reply:
So, let's wait for @alallema to be back for both confirmations

Reply:
So, keeping `http` as it is & shifting docstring to `__init__` .